README.md

```diff
@@ -1,46 +1,103 @@
+To install SecretHub package from NuGet Gallery, run the following command in your project's direcotry: 
+```bash
+dotnet add package SecretHub
+```
+or you can go to the project's `.csproj` file and add the following line:
+```xml
+<PackageReference Include="SecretHub" Version="0.1.0" />
+```
 
-## How to call library functions
-
-- Read
-`SecretHub.SecretVersion secret = SecretHub.Client.Read("path_to_secret");`
-
-- ReadString
-`string secret = SecretHub.Client.ReadString("path_to_secret");`
-
-- Exists
-`bool isSecret = SecretHub.Client.Exists("path_to_secret");`
-
-- Write
-`SecretHub.Client.Write("path_to_secret", "secret_value");`
+## Building from source 
+1. Execute `make nupkg` from the Makefile
+2. Go to your .NET project direcotry and run the following command: `dotnet add package SecretHub -s <path_to_your_secrethub-xgo_repo>`.
 
-- Remove
-`SecretHub.Client.Remove("path_to_secret");`
-
-- Resolve
-`string secret = SecretHub.Client.Resolve("reference");`
-
-- ResolveEnv
-`Doctionary<string, string> resolvedEnv = SecretHub.Client.ResolveEnv(dictionaryToResolve);`
-
-## Example
-
-An example project is found in `azure-example`. There you can see the library in action.
+## How to call library functions
+Before doing any calls to the library, you need to create you SecretHub client. This is done in the following way:
+```csharp
+var client = new SecretHub.Client();
+``` 
+or 
+```csharp
+SecretHub.Client client = new SecretHub.Client();
+```
+
+After you have your client, you can perform on of the following functions:
+
+### `Read(string path)`
+Retrieve a secret, including all its metadata:
+```csharp
+SecretHub.SecretVersion secret = client.Read("path/to/secret");
+Console.WriteLine("The secret value is " + secret.Data);
+```
+`SecretHub.SecretVersion` object represents a version of a secret with sensitive data.
+
+### `ReadString(string path)`
+Retrieve a secret as a string:
+ ```csharp
+ string secret = client.Read("path/to/secret");
+ Console.WriteLine("The secret value is " + secret);
+ ```
+
+### `Exists(string path)`
+Check if a secret exists at `path`:
+```csharp
+bool isSecret = client.Exists("path/to/secret");
+if (isSecret) 
+{
+    Console.WriteLine("The secret exists.");
+} 
+else 
+{
+    Console.WriteLine("The secret does not exists.");
+}
+```
+
+### `Write(string path, string secret)`
+Write a secret containing the contents of `secret` at `path`:
+```csharp
+client.Write("path/to/secret", "secret_value");
+```
+
+### `Remove(string path)`
+Delete the secret found at `path`, if it exists:
+```csharp
+client.Remove("path/to/secret");
+if (!client.Exists("path/to/secret"))
+{
+    Console.WriteLine("Secret deleted successfully");
+}
+```
+
+### `Resolve(string ref)`
+Fetch the value of a secret from SecretHub, when the `ref` has the format `secrethub://<path>`, otherwise it returns `ref` unchanged:
```

Comment:
Yes, that makes it way clearer :ok_hand:

Reply:
Glad to hear that.